index.js

```diff
@@ -138,47 +185,69 @@ function nowServing(line) {
 const line = []
 
 // ***** Scope & Closures - Question 1 *****
-
+const ticketNum = []
+
+function takeATicketNumber(line){
+   
+  let num = ticketNum.length + 1
+  line.push(num)
+  ticketNum.push(num)
+  return `Welcome. You are ticket number ${num}`
+}
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+function ticketNumberGeneratorFunc (){
+  const newticketNum = []
+  
+  return(function(line){
+    let newnum = newticketNum.length + 1
+    line.push(newnum)
+    newticketNum.push(newnum)
+    return `Welcome. You are ticket number ${newnum}`
+  })
+
+}
```

Comment:
Also, good! I have the same comment for this one as in the first scopes & closure question